src/bootstrap/src/core/config/config.rs

```diff
@@ -410,6 +410,217 @@ impl Config {
         flags: Flags,
         get_toml: impl Fn(&Path) -> Result<TomlConfig, toml::de::Error>,
     ) -> Config {
+        let change_id;
+        let bypass_bootstrap_lock;
+        let mut ccache = None;
+        let mut ninja_in_file = true;
+        let mut submodules = None;
+        let mut compiler_docs = false;
+        let mut library_docs_private_items = false;
+        let mut docs_minification = true;
+        let mut docs = true;
+        let mut locked_deps = false;
+        let vendor;
+        let mut target_config = HashMap::new();
+        let mut full_bootstrap = false;
+        let mut bootstrap_cache_path = None;
+        let mut extended = false;
+        let tools;
```

Comment:
What does it mean for a variable to **not** be `mut` here? If the variable is initialized only once, I would just initialize it at the end of the function, in the `Config { ... }` struct literal.